### PR TITLE
COMMERCE-9633 Products published using the generation of all skus combination from options do not appear in the catalog store front

### PR DESCRIPTION
**Pull request by ghost**
https://issues.liferay.com/browse/COMMERCE-9633

### CODE REVIEW

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

File: modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/portlet/action/EditCPInstanceMVCActionCommand.java [hunk 1]
```diff
@@ -219,7 +220,25 @@ private void _buildCPInstances(ActionRequest actionRequest)
 		ServiceContext serviceContext = ServiceContextFactory.getInstance(
 			CPInstance.class.getName(), actionRequest);
 
-		_cpInstanceService.buildCPInstances(cpDefinitionId, serviceContext);
+		List<CPInstance> cpInstances = _cpInstanceService.buildCPInstances(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/portlet/action/EditCPInstanceMVCActionCommand.java:223]: I think this should be put in a callable as well. Can you please review this and advise @alessiorendina @luca-pellizzon , thanks!

**Review by andrea-ale-sbarra — COMMENTED** [review]
Overall I think these changes are fine. Thanks @crescenzoregalf

File: modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/portlet/action/EditCPInstanceMVCActionCommand.java (hunk 1 above)
Comment (luca-pellizzon) [reply modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/portlet/action/EditCPInstanceMVCActionCommand.java:223]: Aren't services already transactional? If so the database should be untouched when any exception occurs.

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by baharturk** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Review by andrea-ale-sbarra — APPROVED** [review]
Thank you @CrescenzoRegaLF and @luca-pellizzon

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Pull request closed without merging**